### PR TITLE
ADD: Job Results

### PR DESCRIPTION
**Pull request by sihrc**
Reviewers: 
----------

- [ ] @mention

What's the purpose of the PR?
-----------------------------

Support Async Job Running in IndicoApi
The way results are handled are kept backwards compatible to original API.

Description of changes
----------------------

Notes to reviewers
------------------

Keep in mind while reviewing code:
----------------------------------
- Is relevant code tested?
- Are added functions/methods documented?
- Separation of concerns (SOC)
- Don't repeat yourself (DRY) 
- Limit the number of positional arguments
- Is function/method length reasonable?
- Can code be broken down into smaller components?
- Where do added functions/methods belong?
- Are variable names descriptive?
- Are errors handled appropriately?
- Can logic be simplified?
- Does the code make sense in context? (expand the diff)

### CODE REVIEW

File: indicoio/utils/api.py [hunk 1]
```diff
@@ -1,38 +1,62 @@
...
+class JobResult(object):
+    def __init__(self, task_id):
+        self.task_id = task_id
+
+    def status(self, api_key=None, cloud=None):
+        return api_handler(
+            None,
+            cloud=cloud,
+            api="async",
+            url_params={"method": f"{self.task_id}/status"},
```
Comment (madisonmay) [inline indicoio/utils/api.py]: Use format instead of f-string if we're going to support py2

**Comment by madisonmay** [comment]
Are we going to use the status endpoint + polling code written external to this for quering results?  What does actually using this downstream look like?

**Comment by sihrc** [comment]
We need to decide whether or not we force the SDK to poll and then fetch results synchronously, or we eventually document that you need to wait for status before getting result.

We can also change the get to wait for status first.
 @madisonmay

**Comment by sihrc** [comment]
TODO:
Add Job Result tests for 

1.  Image Case
2.  Text Case
3.  Custom Collections Predict / Explain Case
4.  PDFExtraction V2 Case

**Comment by sihrc** [comment]
Closing in favor of alternate #280

**Pull request closed without merging**